### PR TITLE
added 'invert' parameter to lwa4d xacro description

### LINKED ISSUE #154: [lwa-4d] Rotation mismatch
Hi, 
I'm seeing a different axis of rotation on the prl+ joints from what the URDF model indicates.  In particular the axis of rotation seems to be pointing toward the base of the robot (positive delta = clockwise rotation) as opposed to what the URDF indicates  -- rotation pointing toward the end effector (positive delta = counter clockwise rotation).  I have checked to make sure my arm is using the default settings.  Are other people having this issue?  Should the model be updated to match the arm or is it better if I flip the direction of the motor in firmware?  Schunk seemed to indicated changing direction in firmware would not be ideal.

### PR DESCRIPTION
**Pull request by athackst**
default value 1 matches DH params, -1 matches default hardware configuration.
fixes #154

### CODE REVIEW

File: schunk_description/urdf/lwa4d/lwa4d.urdf.xacro [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <xacro:include filename="$(find schunk_description)/urdf/lwa4d/lwa4d.gazebo.xacro"/>
   <xacro:include filename="$(find schunk_description)/urdf/lwa4d/lwa4d.transmission.xacro"/>
 
-  <xacro:macro name="schunk_lwa4d" params="parent name *origin has_podest">
+  <xacro:macro name="schunk_lwa4d" params="parent name *origin has_podest invert:=1">
```
Comment (fmessmer) [inline schunk_description/urdf/lwa4d/lwa4d.urdf.xacro]: could we please add a comment for explanation? something like:

```
<!-- invert is a xacro parameter to be used to adapt the urdf according to your hardware setup. use invert:=1 if you configured your hardware to follow DH-convention, i.e. motors have been inverted in the firmware. use invert:=-1 if you are using factory settings, i.e. motors are not inverted in the firmware -->
```

**Comment by fmessmer** [comment]
Thanks, I think this is actually a good way to solve it.
(I was not aware of the possibility to set a default value for xacro parameters using `:=value` :wink: )

Today, I was able to get a hand on our hardware again and I checked the firmware configuration.
Apparently, we inverted the direction of **all** motors (not only the PRL, module_3=joint_1,...,module_9=joint_7)
screenshot_lwa4d_settings

So, I'm wondering what the factory settings for the powerball (joint_6 and joint_7) is...
If the powerball is actually inverted in the factory settings, then that configuration is very inconsistent and messed....then, `inverted` is probably not the best parameter name....maybe we rather use `use_factory_settings` as a boolean parameter and explicitly model the joint axis using `xacro:if`/`xacro:unless`
Can you verify that what the setting of the powerball is in your setup?

**Comment by athackst** [comment]
Hm, my powerball was set up with invert as true.  Although we've had this arm for awhile so who knows what has happened to it over the years.  I've sent an email to Schunk to find out what the factory default actually is.  

Maybe a parameter for each joint would be good since it is configurable?

**Comment by fmessmer** [comment]
Ok, let us know what the default/factory config is once Schunk replied.
I'd like to keep this PR open until then (in order to adjust what is said in the comment - if required).
Is this ok for you?

**Comment by athackst** [comment]
:+1:

**Comment by athackst** [comment]
from Schunk:

PRL+Modules: No/No
ERB145 Modules: No/Yes
ERB115 Modules: Yes/Yes

The LWA4D has PRL+ modules and a ERB115 module, so the default hardware configuration from the manufacturer matches what I had (PRL+: no/no, ERB115: yes/yes).

I updated the xacro to allow both custom configurations and DH/factory switching via use_factory_settings which is false by default.

**Comment by athackst** [comment]
What do you think about this solution?

**Comment by fmessmer** [comment]
Sorry for not getting back earlier...I'm out of the office atm
Looks good already...we'll have our next merge session on Thursday

**Comment by fmessmer** [comment]
tested on our HW (DH-config, i.e. `use_factory_settings=false`, i.e. `invert_j1="false" invert_j2="false" invert_j3="false" invert_j4="false" invert_j5="false" invert_j6="true" invert_j7="true"`)...works as expected...travis is happy...merging!

**Pull request merged**